lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+
+Relaxed placeholders allow matching of everything until a C</> occurs, similar to the regular expression C<([^/]+)>.
+
+  use Mojolicious::Lite;
+
+  # /hello/test
+  # /hello/test.html
+  get '/hello/#you' => 'groovy';
+
+  app->start;
+  __DATA__
+
+  @@ groovy.html.ep
+  Your name is <%= $you %>.
+
+=head2 Wildcard placeholders
+
+Wildcard placeholders allow matching absolutely everything, including C</> and C<.>, similar to the regular expression
+C<(.+)>.
+
+  use Mojolicious::Lite;
+
+  # /hello/test
+  # /hello/test123
+  # /hello/test.123/test/123
+  get '/hello/*you' => 'groovy';
+
+  app->start;
+  __DATA__
+
+  @@ groovy.html.ep
+  Your name is <%= $you %>.
+
+=head2 HTTP methods
+
+Routes can be restricted to specific request methods with different keywords like L<Mojolicious::Lite/"get"> and
+L<Mojolicious::Lite/"any">.
+
+  use Mojolicious::Lite -signatures;
+
+  # GET /hello
+  get '/hello' => sub ($c) {
+    $c->render(text => 'Hello World!');
+  };
+
+  # PUT /hello
+  put '/hello' => sub ($c) {
+    my $size = length $c->req->body;
+    $c->render(text => "You uploaded $size bytes to /hello.");
+  };
+
+  # GET|POST|PATCH /bye
+  any ['GET', 'POST', 'PATCH'] => '/bye' => sub ($c) {
+    $c->render(text => 'Bye World!');
+  };
+
+  # * /whatever
+  any '/whatever' => sub ($c) {
+    my $method = $c->req->method;
+    $c->render(text => "You called /whatever with $method.");
+  };
+
+  app->start;
+
+=head2 Optional placeholders
+
+All placeholders require a value, but by assigning them default values you can make capturing optional.
+
+  use Mojolicious::Lite -signatures;
+
+  # /hello
+  # /hello/Sara
+  get '/hello/:name' => {name => 'Sebastian', day => 'Monday'} => sub ($c) {
+    $c->render(template => 'groovy', format => 'txt');
+  };
+
+  app->start;
+  __DATA__
+
+  @@ groovy.txt.ep
+  My name is <%= $name %> and it is <%= $day %>.
+
+Default values that don't belong to a placeholder simply get merged into the stash all the time.
+
+=head2 Restrictive placeholders
+
+A very easy way to make placeholders more restrictive are alternatives, you just make a list of possible values.
+
+  use Mojolicious::Lite -signatures;
+
+  # /test
+  # /123
+  any '/:foo' => [foo => ['test', '123']] => sub ($c) {
+    my $foo = $c->param('foo');
+    $c->render(text => "Our :foo placeholder matched $foo");
+  };
+
+  app->start;
+
+All placeholders get compiled to a regular expression internally, this process can also be customized. Just make sure
+not to use C<^> and C<$>, or capturing groups C<(...)>, non-capturing groups C<(?:...)> are fine though.
+
+  use Mojolicious::Lite -signatures;
+
+  # /1
+  # /123
+  any '/:bar' => [bar => qr/\d+/] => sub ($c) {
+    my $bar = $c->param('bar');
+    $c->render(text => "Our :bar placeholder matched $bar");
+  };
+
+  app->start;
+
+You can take a closer look at all the generated regular expressions with the command L<Mojolicious::Command::routes>.
+
+  $ ./myapp.pl routes -v
+
+=head2 Under
```

Comment:
Perhaps document here what the auto-generated route name of a route under an under is (i.e. "name" versus "pagename").